### ISSUE TITLE
Why only few columns are visible in hparams at tensorboard

### ISSUE THREAD

**Issue by LIKHITA12** [tensorflow/tensorboard#4770]
Hello, I have trained my model with at least 7 hyperparameters. When I try to visualize these hyperparameters on a tensorboard, I was able to see only a few hyperparameters. I didn't understand why this is happening. When I looked through the source code of Tensorboard, I got to know there is a condition that only few hyperparameters are set to visibility mode and the rest are set to invisible. From the below link I got this code.

[https://github.com/tensorflow/tensorboard/blob/b16ed7cd56ad4e8bee3a0c6cc9ca76042d6e3449/tensorboard/plugins/hparams/tf_hparams_parallel_coords_plot/axes.ts#L419](url)

`// Set the visibility of the remaining axes to false.`
` this._axes.forEach((axis) => {`
 `     if (!visibleColIndices.has(axis.colIndex())) {`
 `       axis.setDisplayed(false);`
 `     }`
 `   });`

Please help me to visualize all hyperparameters on the tensorboard.

**Comment by psybuzz** [comment]
Thank you for the report, and for digging through the code to find that logic!

By default, TensorBoard only shows up to 5 hparams and 5 metrics in the dashboard.  The rest of the hparams will appear in the main table as columns when the checkboxes for those hparams are explicitly checked.

![image](https://user-images.githubusercontent.com/2322480/111019688-c5ed2f00-8375-11eb-95b2-403934104c16.png)

To confirm, are you able to see all your hyperparameters after checking their corresponding checkboxes on the left sidepane?

Googlers, see b/117438608

**Comment by LIKHITA12** [comment]
Actually, I'm using Keras tuner for hyperparameter tuning and visualization in the tensorboard with a package called **kerastuner-tensorboard-logger** and the link is:
[https://pypi.org/project/kerastuner-tensorboard-logger/](url)

I wanted to add another hyperparameter- **batch_size** for my model by using the code below with custom keras-tuner called **MyTuner**:
```
class MyTuner(Hyperband):
def run_trial(self, trial, *args, **kwargs):
   kwargs['batch_size'] = trial.hyperparameters.Int('batch_size', 32, 256, step=16)
   super(MyTuner, self).run_trial(trial, *args, **kwargs)
```

```
tuner = MyTuner(
    create_model,
    objective="val_accuracy",
    max_epochs= 30,
    directory="logs/tuner",
    project_name="hyperparameter_tuning",
    logger=TensorBoardLogger(metrics=['val_accuracy','accuracy'], logdir="logs/hparams")
)
 
```
But, now I'm not able to view the batch-size as a hyperparameter in the tensorboard and got only those hyperparameters that are declared in the model creation.
This is the final output of the tensorboard 
![image](https://user-images.githubusercontent.com/42487404/111022284-d219d080-83f7-11eb-91e7-c872cc40f5c1.png)

**Comment by psybuzz** [comment]
Looking into the KerasTuner code, it looks like `run_trial` reads the hparams directly from `trial.hyperparameters` [1], which should be registered when doing `trial.hyperparameters.Int`, as long as the "conditions" are met.  This requires some more investigation.

[1] https://github.com/keras-team/keras-tuner/blob/master/kerastuner/engine/tuner.py#L301

@omalleyt12 would you be able to help us understand why Keras Tuner may not register the `batch_size` hparam in this example?

**Comment by ConverJens** [comment]
@psybuzz Any update on this?
I'm interested since I'm running keras tuner in the Tuner component in TFX and I haven't got hparams to work at all in Tensorboard with this setup.

**Comment by ConverJens** [comment]
@haifeng-jin Sorry for the late response! This actually worked when I updated Keras tuner to 1.0.2.

**Comment by haifeng-jin** [comment]
@ConverJens Thank you for the update! I will soon have a tutorial for how to visualize the tuning process with TensorBoard. Hope that would help more people using it.
@psybuzz I guess we are good to close the issue now.

**Comment by psybuzz** [comment]
Neat, a tutorial covering TB hparams with Keras Tuner would be great.  For other readers, please leave a comment if you observe this after upgrading to Keras Tuner 1.0.2.

**Comment by haifeng-jin** [comment]
A quick update. If you are not seeing all the hyperparams, make sure to scroll down the panel on the left to check the check-boxes.
By default tensorboard will only show 4 hyperparams.

![Screenshot 2021-06-22 2 30 17 PM](https://user-images.githubusercontent.com/5476582/122987909-c464d700-d366-11eb-8ef0-6f97a1c0f5cf.png)
